sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +554,257 @@ def test_param_sampler():
+        # take weighted average and weighted std
+        expected_mean = 1 * 1. / 4. + 1. / 3. * 3. / 4.
+        expected_std = np.sqrt(1. / 4 * (expected_mean - 1) ** 2 +
+                               3. / 4 * (expected_mean - 1. / 3.) ** 2)
+        assert_almost_equal(mean, expected_mean)
+        assert_almost_equal(std, expected_std)
+
+    # once with iid=False
+    grid_search = GridSearchCV(SVC(),
+                               param_grid={'C': [1, 10]},
+                               cv=cv, iid=False)
+    random_search = RandomizedSearchCV(SVC(), n_iter=2,
+                                       param_distributions={'C': [1, 10]},
+                                       cv=cv, iid=False)
+
+    for search in (grid_search, random_search):
+        search.fit(X, y)
+        assert_false(search.iid)
+
+        cv_scores, mean, std = search._get_candidate_scores(0)
+        assert_equal(search.results_['param_C'][0], 1)
+        # scores are the same as above
+        assert_array_almost_equal(cv_scores, [1, 1. / 3.])
+        # Unweighted mean/std is used
+        assert_almost_equal(mean, np.mean(cv_scores))
+        assert_almost_equal(std, np.std(cv_scores))
+
+
+def test_search_results_rank_tie_breaking():
+    X, y = make_blobs(n_samples=50)
+
+    # The two C values are close enough to give similar models
+    # which would result in a tie of their mean cv-scores
+    param_grid = {'C': [1, 1.001, 0.001]}
+
+    grid_search = GridSearchCV(SVC(), param_grid=param_grid)
+    random_search = RandomizedSearchCV(SVC(), n_iter=3,
+                                       param_distributions=param_grid)
+
+    for search in (grid_search, random_search):
+        search.fit(X, y)
+        results = search.results_
+        # Check tie breaking strategy -
+        # Check that there is a tie in the mean scores btw
+        # candidates 1 and 2 alone
+        assert_almost_equal(results['test_mean_score'][0],
+                            results['test_mean_score'][1])
+        try:
+            assert_almost_equal(results['test_mean_score'][1],
+                                results['test_mean_score'][2])
+        except AssertionError:
+            pass
+        # 'min' rank should be assigned to the tied candidates
+        assert_almost_equal(search.results_['test_rank_score'], [1, 1, 3])
+
+
+def test_search_results_none_param():
+    X, y = [[1], [2], [3], [4], [5]], [0, 0, 0, 0, 1]
+    estimators = (DecisionTreeRegressor(), DecisionTreeClassifier())
+    est_parameters = {"random_state": [0, None]}
+    cv = KFold(random_state=0)
+
+    for est in estimators:
+        grid_search = GridSearchCV(est, est_parameters, cv=cv).fit(X, y)
+        res_params = grid_search.results_['params']
+        for cand_i in range(len(res_params)):
+            est.set_params(**res_params[cand_i])
+            assert_equal(est.random_state,
+                         grid_search.results_['param_random_state'][cand_i])
```

Comment:
You should check one of these to an explicit expected value, something like `assert_array_equal(grid_search.results_['param_random_state'], [0, None])`

It can be possible that the code is written in such a way that both `res_params` and `grid_search.results_['param_random_state']` is wrong.